api/app/Http/Controllers/BaseController.php

```diff
@@ -34,61 +34,157 @@
     protected $transformer = 'App\Http\Transformers\BaseTransformer';
 
     /**
-     * Transform a collection for response.
+     * Get all entities.
      *
-     * @param  Collection $collection
-     * @return array
+     * @return Response
      */
-    protected function collection(Collection $collection)
+    public function getAll()
     {
-        $transformer = App::make('App\Services\Transformer');
+        return $this->collection($this->repository->all());
+    }
 
-        return $transformer->collection($collection, new $this->transformer);
+    /**
+     * Get one entity.
+     *
+     * @param  string  $id
+     * @return Response
+     */
+    public function getOne($id)
+    {
+        return $this->item($this->repository->find($id));
     }
 
     /**
-     * Transform an item for response.
+     * Post a new entity.
      *
-     * @param  Model  $item
-     * @return array
+     * @param  Request $request
+     * @return mixed
      */
-    protected function item(Model $item)
+    public function postOne(Request $request)
     {
-        $transformer = App::make('App\Services\Transformer');
+        $this->validator->with($request->all())->validate();
 
-        return $transformer->item($item, new $this->transformer);
+        return response($this->repository->create($request->all()), 201);
```

Comment:
Is this response going to be transformed? Should the line not be 
```
return response($this->item($this->repository->create($request->all())), 201);
```
If thats the case we should probably extend the functionality of `$this->item` to include status code as an optional param (200 being default) as the above is pretty unwieldy.